### PR TITLE
sign & verify in bitcoin and ethereum networks

### PR DESCRIPTION
**Pull request by ghost**
before the select functionality

### CODE REVIEW

File: packages/suite/src/actions/wallet/signVerifyActions.ts [hunk 1]
```diff
@@ -58,50 +84,72 @@ export const verify = (address: string, message: string, signature: string, hex
...
+
+    let fn;
+    const params = {
+        address,
+        message,
+        signature,
+        coin: account.symbol,
+        hex,
+    };
+    const error = isAddressValid(address, params.coin as Account$['symbol'])
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/signVerifyActions.ts]: use `account.symbol` instead of `params.coin as Account$['symbol']`

File: packages/suite/src/actions/wallet/signVerifyActions.ts [hunk 2]
```diff
@@ -58,50 +84,72 @@ export const verify = (address: string, message: string, signature: string, hex
...
+                payload: {
+                    error: `Unsupported network: ${account.networkType}`,
+                    code: undefined,
+                    signature: '',
+                },
+            });
+            break;
+    }
 
     if (error) {
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/signVerifyActions.ts]: move this block to line ~102, before you declare fn and params

Comment (szymonlesisz) [reply packages/suite/src/actions/wallet/signVerifyActions.ts]: add 2, is this double-validation needed at here? you've already validate address in `inputChange` therefore you shouldn't be able to call `verify` with invalid address

File: packages/suite/src/config/wallet/menu.tsx [hunk 3]
```diff
@@ -35,9 +35,11 @@ export const ITEMS = [
         title: <Translation id="TR_NAV_SIGN_AND_VERIFY" />,
         icon: 'SIGN',
         isHidden: (_account: Account) => {
-            return true;
-            // const network = NETWORKS.find(n => n.symbol === account.symbol);
-            // return !(network && network.testnet && network.hasSignVerify);
+            const network = NETWORKS.find(n => n.symbol === _account.symbol);
+            return !(
+                (network?.networkType === 'bitcoin' || network?.networkType === 'ethereum') &&
```
Comment (szymonlesisz) [inline packages/suite/src/config/wallet/menu.tsx]: `return !(network && network.testnet && network.hasSignVerify)` vas valid 
`network?.networkType === 'bitcoin' || network?.networkType === 'ethereum'` is unnecessary

File: packages/suite/src/views/wallet/sign-verify/index.tsx [hunk 4]
```diff
@@ -74,155 +76,190 @@ interface FormEvent {
...
+    } = props;
+
+    useEffect(() => {
+        return () => {
+            signVerifyActions.clearSign(); // clear inputs after coin change
+            signVerifyActions.clearVerify();
+        };
+    }, [signVerifyActions]);
+
+    const { device } = useDevice();
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/index.tsx]: const { device, isLocked } = useDevice();

use `isLocked` function instead of `!device.connected` (disabling buttons)

File: packages/suite/src/actions/wallet/signVerifyActions.ts [hunk 5]
```diff
@@ -113,7 +161,12 @@ export const inputChange = (inputName: inputNameType, value: string) => (dispatc
     });
 
     if (inputName === 'verifyAddress') {
-        const error = validateAddress(value);
+        const { account } = getState().wallet.selectedAccount;
+        if (!account) return;
+        const error = isAddressValid(value, account.symbol as Account$['symbol'])
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/signVerifyActions.ts]: account.symbol is enough, no need to retype it

File: packages/suite/src/views/wallet/sign-verify/index.tsx [hunk 6]
```diff
@@ -74,155 +76,190 @@ interface FormEvent {
...
-                </Wrapper>
-            </WalletLayout>
-        );
-    }
-}
+    const handleInputChange = (event: FormEvent) => {
+        signVerifyActions.inputChange(event.target.name as InputNameType[0], event.target.value);
+    };
+
+    const getAddress = (): AddressToSign => {
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/index.tsx]: `AddressToSign` is unnecessary, this method will always return object { address, path } therefore additional typings is not required

File: packages/suite/src/views/wallet/sign-verify/index.tsx [hunk 7]
```diff
@@ -74,155 +76,190 @@ interface FormEvent {
...
+        }
+
+        return {
+            address: '',
+            path: '',
+        };
+    };
+
+    const verifyAddressError = getError('verifyAddress');
+
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/index.tsx]: consider to call `getAddress()` once here and use it in lower scope (right now you're calling same function multiple times

File: packages/suite/src/config/wallet/menu.tsx [hunk 8]
```diff
@@ -35,9 +35,8 @@ export const ITEMS = [
         title: <Translation id="TR_NAV_SIGN_AND_VERIFY" />,
         icon: 'SIGN',
         isHidden: (_account: Account) => {
-            return true;
-            // const network = NETWORKS.find(n => n.symbol === account.symbol);
-            // return !(network && network.testnet && network.hasSignVerify);
+            const network = NETWORKS.find(n => n.symbol === _account.symbol);
+            return !(network && network.hasSignVerify);
```
Comment (szymonlesisz) [inline packages/suite/src/config/wallet/menu.tsx:39]: `return network && network.hasSignVerify` :/

Comment (szymonlesisz) [reply packages/suite/src/config/wallet/menu.tsx:39]: my mistake it needs to return "true" for hidden (reversed logic)

File: packages/suite/src/views/wallet/sign-verify/index.tsx [hunk 9]
```diff
@@ -74,155 +73,187 @@ interface FormEvent {
...
-}
+    const handleInputChange = (event: FormEvent) => {
+        signVerifyActions.inputChange(event.target.name as InputNameType[0], event.target.value);
+    };
+
+    const getAddress = () => {
+        if (
+            selectedAccount.account.networkType === 'bitcoin' &&
+            typeof selectedAccount.account.addresses?.unused[0]?.address === 'string' &&
+            typeof selectedAccount.account.addresses?.unused[0]?.address === 'string'
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/index.tsx]: why `typeof selectedAccount.account.addresses?.unused[0]?.address === 'string'` twice?

Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/index.tsx]: it is mistake, path must be on this place

File: packages/suite/src/utils/suite/dom.ts [hunk 10]
```diff
@@ -50,6 +50,9 @@ export const useKeyPress = (targetKey: string) => {
 
 export const copyToClipboard = (value: string, parent: HTMLDivElement | HTMLPreElement | null) => {
     try {
+        if (value.length <= 0) {
+            return false;
```
Comment (szymonlesisz) [inline packages/suite/src/utils/suite/dom.ts]: you should throw Error instead of return false, error will be catch in lower block

File: packages/suite/src/actions/wallet/signVerifyActions.ts [hunk 11]
```diff
@@ -1,133 +0,0 @@
-import TrezorConnect from 'trezor-connect';
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/signVerifyActions.ts:1]: i dont understand why did you move all actions to component?

File: packages/suite/src/views/wallet/sign-verify/Container.ts [hunk 12]
```diff
@@ -1,21 +1,28 @@
...
 
 const mapStateToProps = (state: AppState) => ({
     selectedAccount: state.wallet.selectedAccount,
+    suite: state.suite,
     signVerify: state.wallet.signVerify,
 });
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-    signVerifyActions: bindActionCreators(signVerifyActions, dispatch),
+    notificationActions: bindActionCreators(notificationActions, dispatch),
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/Container.ts]: mapDispatchToProps is unnecessary, you can use `useActions` hook instead

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+                            type="text"
+                            innerRef={register}
+                            readOnly
+                        />
+                    )}
+                    <StyledIcon
+                        icon="COPY"
+                        onClick={() => {
+                            // eslint-disable-next-line no-unused-expressions
+                            copyToClipboard(address, null)
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx]: instead of validation address.length inside `copyToClipboard` and adding more logic you should disable this button (do not allow to copy if address is empty or invalid)

File: packages/suite/src/views/wallet/sign-verify/index.tsx [hunk 14]
```diff
@@ -1,228 +1,59 @@
...
-    }
-}
+    return (
+        <WalletLayout title="Sign & Verify" account={selectedAccount}>
+            <SignVerifyContainer>
+                <SignMessage
+                    notificationActions={notificationActions}
+                    account={selectedAccount.account}
+                    isLocked={isLocked}
+                    device={suite.device}
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/index.tsx]: you can get device object from `useDevice` hook instead of adding `props.suite`

in fact you'll not need `device` here at all if you move back actions to where they belong

File: packages/suite/src/views/wallet/sign-verify/index.tsx [hunk 15]
```diff
@@ -1,228 +1,45 @@
...
-    @media screen and (max-width: ${variables.SCREEN_SIZE.MD}) {
-        padding-left: 0px;
+    flex-direction: row;
+    @media screen and (max-width: ${variables.SCREEN_SIZE.SM}) {
+        flex-direction: column;
     }
 `;
 
-interface Props extends WrappedComponentProps {
+interface Props {
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/index.tsx]: this is the same as `StateProps`, unnecessary declaration

File: packages/suite/src/actions/wallet/signVerifyActions.ts [hunk 16]
```diff
@@ -1,133 +1,130 @@
...
 
-export const verify = (address: string, message: string, signature: string, hex = false) => async (
+export const sign = (message: string, path: string, hex = false) => async (
     dispatch: Dispatch,
     getState: GetState,
-) => {
-    const selectedDevice = getState().suite.device;
-    if (!selectedDevice) return;
-    const error = validateAddress(address);
+): Promise<string | boolean> => {
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/signVerifyActions.ts]: you don't have to type it, ts should do it automatically

nitpick: you dont have to return boolean since you're checking if response is a `typeof string`. you can just `return` (void)

File: packages/suite/src/actions/wallet/signVerifyActions.ts [hunk 17]
```diff
@@ -1,133 +1,130 @@
...
-                instance: selectedDevice.instance,
-                state: selectedDevice.state,
-            },
-            address,
-            message,
-            signature,
-            hex,
-            useEmptyPassphrase: selectedDevice.useEmptyPassphrase,
-        });
+    const params = {
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/signVerifyActions.ts:37]: add `{
  device,
  useEmptyPassphrase: device.useEmptyPassphrase,
}` to params

File: packages/suite/src/actions/wallet/signVerifyActions.ts [hunk 18]
```diff
@@ -1,133 +1,130 @@
...
+    signature: string,
+    hex = false,
+) => async (dispatch: Dispatch, getState: GetState) => {
+    const { device } = getState().suite;
+    const { account } = getState().wallet.selectedAccount;
+    if (!device || !account) return;
+
+    let fn;
+    const params = {
+        address,
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/signVerifyActions.ts:72]: add { device, useEmptyPassphrase: device.useEmptyPassphrase, } to params

File: packages/suite/src/utils/suite/dom.ts [hunk 19]
```diff
@@ -50,6 +50,9 @@ export const useKeyPress = (targetKey: string) => {
 
 export const copyToClipboard = (value: string, parent: HTMLDivElement | HTMLPreElement | null) => {
     try {
+        if (value.length <= 0) {
```
Comment (szymonlesisz) [inline packages/suite/src/utils/suite/dom.ts]: remove this, disable button instead

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx [hunk 20]
```diff
@@ -0,0 +1,235 @@
...
+                        <Input
+                            topLabel={<Translation id="TR_ADDRESS" />}
+                            name="address"
+                            value={address}
+                            type="text"
+                            innerRef={register}
+                            readOnly
+                        />
+                    )}
+                    <StyledIcon
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx]: wrap it into a button with icon, and set `isDisabled` when empty address (move validation from `copyToClipboard`

File: packages/suite/src/actions/wallet/signVerifyActions.ts [hunk 21]
```diff
@@ -1,133 +1,130 @@
...
-    });
-
-    if (response && response.success) {
-        dispatch({
-            type: SIGN_VERIFY.SIGN_SUCCESS,
-            signSignature: response.payload.signature,
-        });
-    } else {
+    if (typeof errorResponse === 'string') {
         dispatch(
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/signVerifyActions.ts]: you dont have to handle this error, just do nothing (like other implementations of `copyToClipboard`

File: packages/suite/src/views/wallet/sign-verify/components/VerifyMessage/index.tsx [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+    margin-left: 10px;
+
+    &:first-child {
+        margin-left: 0;
+    }
+`;
+type errorsMessage = 'TR_ADDRESS_IS_NOT_VALID' | 'TR_REQUIRED_FIELD';
+
+const VerifyMessage = ({ account, isLocked }: Props) => {
+    const { initVerify } = useActions({ initVerify: signVerifyActions.initVerify });
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/components/VerifyMessage/index.tsx]: why 'init' prefix? should be just `verify`, no?

Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/components/VerifyMessage/index.tsx]: too much verify.... variables on page

File: packages/suite/src/utils/suite/dom.ts (hunk 19 above)
Comment (ghost) [reply packages/suite/src/utils/suite/dom.ts]: I believe that the user should not be allowed to copy an empty string, he may lose money due to this behavior

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx (hunk 20 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx]: the user does not expect to see a toast message about successful copying, so user can click on the disabled button, I suggest discussing it with the designer

File: packages/suite/src/actions/wallet/signVerifyActions.ts (hunk 21 above)
Comment (ghost) [reply packages/suite/src/actions/wallet/signVerifyActions.ts]: same as above

File: packages/suite/src/actions/wallet/signVerifyActions.ts (hunk 17 above)
Comment (szymonlesisz) [reply packages/suite/src/actions/wallet/signVerifyActions.ts:37]: still missing

File: packages/suite/src/support/messages.ts [hunk 23]
```diff
@@ -533,6 +541,10 @@ const definedMessages = defineMessages({
         defaultMessage: 'Copy to clipboard',
         id: 'TR_COPY_TO_CLIPBOARD',
     },
+    TR_COPY_TO_CLIPBOARD_FAILED: {
```
Comment (szymonlesisz) [inline packages/suite/src/support/messages.ts]: not used

File: packages/integration-tests/projects/suite-web/tests/browser/edge.test.ts [hunk 24]
```diff
@@ -1,4 +1,4 @@
-// @beta
+// @stable
```
Comment (mroz22) [inline packages/integration-tests/projects/suite-web/tests/browser/edge.test.ts:1]: this actually should not be part of this branch, you have probably made something wrong with git (regarding merge or rebase) would be best to drop 63a979021195e965228f4ad0392289f098955e1c and b12a086268f6604c0451baa6a8bb2472c507682e. My favourite way to do this is to do local rebase, something like `git rebase -i HEAD~22` and then use drop for commits you dont want to have in your branch.

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx [hunk 25]
```diff
@@ -0,0 +1,249 @@
...
+        };
+    }, [reset]);
+
+    return (
+        <Card title={<Translation id="TR_SIGN_MESSAGE" />} isColumn>
+            <Column>
+                <Row>
+                    {account.networkType === 'bitcoin' && (
+                        <Controller
+                            as={<StyledSelect options={accountsSelectOpt} />}
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:129]: In bitcoin case (got multiple options) `Select` should have preselected first unused address
There is no possibility to have "empty" address field in select. You can only sign message with your own address/path (and we know them all), there is no other way to do this...

File: packages/integration-tests/projects/suite-web/tests/browser/edge.test.ts (hunk 24 above)
Comment (ghost) [reply packages/integration-tests/projects/suite-web/tests/browser/edge.test.ts:1]: I don't have these commits in the branch in the local branch (and pull "up to date")
Screenshot 2020-06-18 at 17 38 09

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx [hunk 26]
```diff
@@ -0,0 +1,249 @@
...
+                                    TR_REQUIRED_FIELD: (selectValue: Inputs['accountsSelect']) => {
+                                        return selectValue.value.length > 0;
+                                    },
+                                },
+                            }}
+                        />
+                    )}
+                    {/* input is more convenient from the ability to select and copy */}
+                    {account.networkType === 'ethereum' && (
+                        <Input
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:157]: use disabled `Select` with one option. You shouldn't be able to copy address unverified  by device

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx [hunk 27]
```diff
@@ -0,0 +1,249 @@
...
+    address: string;
+    path: string;
+    signMessage: string;
+    accountsSelect: { label: string; value: string; type: '' };
+    signature: string;
+};
+
+let path: AddressToSign['address'] = ''; // fast click on copy button failed if put this inside a component
+let address: AddressToSign['path'] = '';
+let accountsSelectOpt: AddressToSign['accountsSelectOpt'];
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:78]: `path`, `address` and `accountsSelectOpt` should be a part of `SignMessage` component state (useState hook)
keeping them as out-of-scope variables is anti pattern

In fact the only thing you need to work with is a `Select` currently selected option (no need to split it into 3 separated variables)

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx (hunk 25 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:129]: I don't think it's obvious, I didn't have a exactly described task, ok I'll fix it

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx (hunk 26 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:157]: input was left there because of the old feature, where you can copy the address, ok i rewrite

File: packages/suite/src/components/suite/modals/SignedMessage/index.tsx [hunk 28]
```diff
@@ -0,0 +1,129 @@
...
+    const copy = () => {
+        const result = copyToClipboard(signedTextObj.text, htmlElement.current);
+        if (typeof result !== 'string') {
+            addNotification({ type: 'copy-to-clipboard' });
+        }
+    };
+
+    useEffect(copy, []); // componentDidMount, copy at first appear
+
+    const download = () => {
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/modals/SignedMessage/index.tsx:68]: Instead of copy/pasting this function from `Log` modal move this to `@suite-utils/dom`, right next to `copyToClipboard`
and then use it in both places, here and `Log` modal

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx (hunk 27 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:78]: OK, anti-pattern, I've been writing in react since may 25, less than month, i will fix it

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx [hunk 29]
```diff
@@ -0,0 +1,249 @@
...
+            <Column>
+                <Row>
+                    {account.networkType === 'bitcoin' && (
+                        <Controller
+                            as={<StyledSelect options={accountsSelectOpt} />}
+                            control={control}
+                            topLabel={<Translation id="TR_ADDRESS" />}
+                            name="accountsSelect"
+                            state={errors.accountsSelect ? 'error' : undefined}
+                            onChange={([selected]) => {
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:134]: i think changing option (address) should clear signature (and copy button), otherwise result will be inconsistent.
Example:
1. sign (signature is generated)
2. change address (pick different)
3. then click "Copy result" (address displayed in modal is not the one you have used to sign, the result you now have in clipboard is incorrect)

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx (hunk 25 above)
Comment (szymonlesisz) [reply packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:129]: i'm pretty sure that i've told you that multiple times...

File: packages/suite/src/actions/wallet/signVerifyActions.ts (hunk 16 above)
Comment (ghost) [reply packages/suite/src/actions/wallet/signVerifyActions.ts]: Yes, you can write this, and make from typescript -> javaScript, anyway I rewrote it

File: packages/suite/src/actions/wallet/signVerifyActions.ts (hunk 17 above)
Comment (ghost) [reply packages/suite/src/actions/wallet/signVerifyActions.ts:37]: this was not described in the API, psychics on vacation, but i rewrote this

File: packages/suite/src/utils/suite/dom.ts (hunk 19 above)
Comment (ghost) [reply packages/suite/src/utils/suite/dom.ts]: anti-pattern in fintech, possible wrong bug, but i rewrote this

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx (hunk 25 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:129]: "In bitcoin case (got multiple options) Select should have preselected first unused address" - no one told me that, and it's bad form to give voice tasks in IT and then claim that you said it

File: packages/suite/src/components/suite/modals/SignedMessage/index.tsx (hunk 28 above)
Comment (ghost) [reply packages/suite/src/components/suite/modals/SignedMessage/index.tsx:68]: I did not consider it necessary to rewrite someone else's component, which I do not know exactly where it is used and when, I have been on this project for less than a month, you now wrote - OK, I can do refactoring

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx (hunk 29 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx:134]: agree

File: packages/suite/src/support/messages.ts [hunk 30]
```diff
@@ -1170,6 +1178,27 @@ const definedMessages = defineMessages({
         description: 'Description of Trezor model T',
         id: 'TR_MODEL_T_DESC',
     },
+    TR_SUCCESS_COPIED: {
```
Comment (szymonlesisz) [inline packages/suite/src/support/messages.ts:1181]: use existing `TOAST_COPY_TO_CLIPBOARD` instead

File: packages/suite/src/actions/wallet/signVerifyActions.ts (hunk 1 above)
Comment (ghost) [reply packages/suite/src/actions/wallet/signVerifyActions.ts]: I couldn't have known this, this is the first serious component in my work on this project, i rewrote this

File: packages/suite/src/support/messages.ts (hunk 30 above)
Comment (szymonlesisz) [inline packages/suite/src/support/messages.ts:1181]: use existing `TOAST_COPY_TO_CLIPBOARD` instead

File: packages/suite/src/actions/wallet/signVerifyActions.ts (hunk 2 above)
Comment (ghost) [reply packages/suite/src/actions/wallet/signVerifyActions.ts]: double validation is ok in finance, but ok, I rewritten it

File: packages/suite/src/views/wallet/sign-verify/index.tsx (hunk 4 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/index.tsx]: again, I can't know this in a couple of weekы (these are project specific things), but well I rewritten it (I'm writing this to the fact that you said you wrote me too many comments on the pull request)

File: packages/suite/src/views/wallet/sign-verify/index.tsx (hunk 6 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/index.tsx]: you confused me, I thought we were writing in typescript (nor javascript), and you say that sometimes typing is not necessary (?); when it is necessary, and when it is not necessary in your subjective opinion, no one explained to me

File: packages/suite/src/views/wallet/sign-verify/index.tsx (hunk 7 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/index.tsx]: actually, in fact, through the hook that you told me to use, I had to get some values also through function calls (getValues () ), but OK, I rewritten it

File: packages/suite/src/utils/suite/dom.ts (hunk 10 above)
Comment (ghost) [reply packages/suite/src/utils/suite/dom.ts]: throwing an error and then in the catch block return error.message as a string is anti-pattern, but ok, i rewrote

File: packages/suite/src/views/wallet/sign-verify/components/VerifyMessage/index.tsx [hunk 31]
```diff
@@ -0,0 +1,161 @@
...
+                    />
+                </Row>
+                <RowButtons>
+                    <StyledButton onClick={() => reset()} variant="secondary">
+                        <Translation id="TR_CLEAR" />
+                    </StyledButton>
+                    <StyledButton
+                        isDisabled={!isValid || isLocked()}
+                        onClick={async () => {
+                            await verify(
```
Comment (szymonlesisz) [inline packages/suite/src/views/wallet/sign-verify/components/VerifyMessage/index.tsx:147]: this "async/await" is unnecessary here since you don't work with the result
`onClick={() => verify(...)}` is enough

File: packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx (hunk 13 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/components/SignMessage/index.tsx]: as I said already, the user does not expect toast messages about successful copying and the button color is gray, the user does not have a reference for comparison this is a disabled button or not, so this decision is confusing in terms of ui, but I rewritten as you wanted

File: packages/suite/src/views/wallet/sign-verify/components/VerifyMessage/index.tsx (hunk 31 above)
Comment (ghost) [reply packages/suite/src/views/wallet/sign-verify/components/VerifyMessage/index.tsx:147]: it's weird, not to write a canonical function call, it will work the same way in this case; but okay, I'll rewrite it

**Comment by keraf** [comment]
@prusnak Do you know what's the status of this PR? Should we look into fixing it or reject the change?

**Comment by prusnak** [comment]
@keraf I think this is a question for the product - do we want to have this functionality in the Suite? If yes, let's prioritize it and find someone who will address all outstanding comments from Szymi, do the rebase, cleanup, etc.
